src/qibo/backends/__init__.py

```diff
@@ -22,25 +22,37 @@ def __init__(self):
         self._availability = {}
 
         # create numpy backend (is always available as numpy is a requirement)
-        if self.check_availability("numpy"):
+        if self.check_availability("numpy", check_version=False):
             from qibo.backends.numpy import NumpyBackend
             self.qnp = NumpyBackend()
         else:  # pragma: no cover
             raise_error(ModuleNotFoundError, "Numpy is not installed. "
                                              "Please install it using "
                                              "`pip install numpy`.")
 
+        # loading backend names and version
+        self._backends_min_version = {}
+        for backend in self.profile.get("backends"):
+            self._backends_min_version[backend.get("name")] = backend.get("minimum_version")
+
         # find the default backend name
         default_backend = os.environ.get('QIBO_BACKEND', self.profile.get('default'))
-        # change the default backend if it is not available
-        if not self.check_availability(default_backend):  # pragma: no cover
-            for backend in self.profile.get('backends'):
-                name = backend.get('name')
-                if self.check_availability(name):
-                    default_backend = name
-                    break
-                # make numpy default if no other backend is available
-                default_backend = "numpy"
+
+        # check if default backend is described in the profile file
+        if default_backend != "numpy":
+
+            if default_backend not in (backend.get("name") for backend in self.profile.get("backends")): # pragma: no cover
+                raise_error(ModuleNotFoundError, f"Default backend {default_backend} not set in {profile_path}.")
+
+            # change the default backend if it is not available
+            if not self.check_availability(default_backend):  # pragma: no cover
+                for backend in self.profile.get('backends'):
```

Comment:
I should have done this in a previous PR but perhaps we could add a comment here explaining that this loop sets as default backend the first available in the order the backends are defined in profiles.yml and if none of these backends is available then we fall back to numpy.

Reply:
Thanks, fixed.